### PR TITLE
Entity cache: reorganize subgraph configuration override

### PR DESCRIPTION
**Pull request by Geal**
**breaking change** this updates the entity cache configuration to reuse the all/subgraphs pattern of configuration override used in other parts of the router.

Since this is a preview feature, do we still make a configuration migration for it?

---

**Checklist**

Complete the checklist (and note appropriate exceptions) before the PR is marked ready-for-review.

- [ ] Changes are compatible[^1]
- [ ] Documentation[^2] completed
- [ ] Performance impact assessed and acceptable
- Tests added and passing[^3]
    - [ ] Unit Tests
    - [ ] Integration Tests
    - [ ] Manual Tests

**Exceptions**

*Note any exceptions here*

**Notes**

[^1]: It may be appropriate to bring upcoming changes to the attention of other (impacted) groups. Please endeavour to do this before seeking PR approval. The mechanism for doing this will vary considerably, so use your judgement as to how and when to do this.
[^2]: Configuration is an important part of many changes. Where applicable please try to document configuration examples.
[^3]: Tick whichever testing boxes are applicable. If you are adding Manual Tests, please document the manual testing (extensively) in the Exceptions.

[ROUTER-369]: https://apollographql.atlassian.net/browse/ROUTER-369?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by bnjjj — APPROVED** [review]
Missing a changelog, but otherwise LGTM

**Review by garypen — APPROVED** [review]

**Review by IvanGoncharov — APPROVED** [review]

File: apollo-router/src/plugins/cache/entity.rs [hunk 1]
```diff
@@ -67,34 +68,30 @@ pub(crate) struct Config {
     /// activates caching for all subgraphs, unless overriden in subgraph specific configuration
```
Comment (BrynCooke) [inline apollo-router/src/plugins/cache/entity.rs:68]:
```suggestion
    /// Enable or disable the the entity caching feature
```

**Review by BrynCooke — CHANGES_REQUESTED** [review]

**Comment by BrynCooke** [comment]
@Geal Can you make a generic struct to hold `all` and `subgraphs` that can be reused? You've mentioned before that we should have a standard way of doing this stuff and this would help us.

**Comment by Geal** [comment]
> @Geal Can you make a generic struct to hold all and subgraphs that can be reused? You've mentioned before that we should have a standard way of doing this stuff and this would help us.
There is, it is called `SubgraphConfiguration`, it is used in this PR and defined here: https://github.com/apollographql/router/blob/dd9005e5cdca5090af049f2d236e3a3a47a89e34/apollo-router/src/configuration/subgraph.rs#L78-L88

**Comment by Geal** [comment]
I made the following changes in [bfb7fbe](https://github.com/apollographql/router/pull/5474/commits/bfb7fbe0dd81be6886873c9bd8b5a3b7a5d08e2a) and [d36870d](https://github.com/apollographql/router/pull/5474/commits/d36870d968cb1b29209cdc1253ac010c0e158e57):
- the top level `enabled` is explicitely a precondition to enabling entity cache on any subgraph
- uses the `SubgraphConfiguration::get` method to fall back to the `all` configuration if the subgraph is not present (per field override between `all` and `subgraphs` is done in the `Deserialize` implementation)

**Review by BrynCooke — APPROVED** [review]

**Pull request merged**